Blog: Hacktoberfest 2022 (#1320)
zaproxy/zaproxy-website: site/content/blog/2022-10-01-zap-and-hacktoberfest/index.md

---
title: "ZAP and Hacktoberfest"
summary: >
    ZAP is partcipating in Hacktoberfest 2022.
images:
- https://www.zaproxy.org/site/blog/2022-10-01-zap-and-hacktoberfest/images/Hfest-Badge-2-Color-Void@2x.png
type: post
tags:
- blog
- hacktoberfest
date: "2022-10-01"
addSocialPreview: true
authors:
- arkaprabha
---
![Hacktoberfest X ZAP](images/zap_Hacktoberfest.png)

> We are happy to announce that ZAP is partcipating in Hacktoberfest 2022. Hacktoberfest 2022 is a month-long celebration of open source software presented by [DigitalOcean](https://digitalocean.com/), [Appwrite](https://appwrite.io/) and [Docker](https://docker.com/). Hacktoberfest is open to everyone in our community! You can sign up anytime between October 1 and October 31.

Reviewer: - Typo in participating.
 - Link to the official site instead.

Author: official site? of hacktoberfest?

Author: typo fixed

Reviewer: Yes but link to that site instead of the others, we don't need to do the marketing, already done in the site.

## Contributors
This is what contributors need to know to participate and complete Hacktoberfest 2022:
- Register anytime between September 26 and October 31.

> - Contributions/pull requests can be made to any repositories under the [OWASP ZAP](https://github.com/zaproxy) organization.

Reviewer: This seems to be an invitation to unwanted pull requests, not all repositories should be changed.

Author: Okay so just zaproxy, zap-extensions and zap-hud?

Author: I thought community scripts can be added too :)

Reviewer: The ones linked from https://www.zaproxy.org/docs/developer/quick-start-build/ and Community Scripts sounds good.

Author: I'll add a bullet list

Author: added the list

- Read the [ZAP Contributing Guide](https://www.zaproxy.org/docs/contribute/).
- We recommend you to go through [ZAP Development guide](https://www.zaproxy.org/docs/developer/), it has all the answers to your questions about the development environment setup and the verifications you must perform before your PR is merged. 
- If you are beginner, we recommend you to try out the curated [hacktoberfest issues](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3AHacktoberFest) we have for you. 
- If you would like to tackle a bit more challenging ones, you can check out the [good second issues](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+second+issue%22) and [enhancement](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) issues.
- Have 4 pull requests accepted between October 1 and October 31.

> - The first 40,000 participants (maintainers and contributors) who successfully complete Hacktoberfest can elect to recieve one out of two prizes: a tree planted in their name, or the Hacktoberfest 2022 t-shirt. 

Reviewer: Typo in receive.

---
title: "ZAP and Hacktoberfest 2022"
summary: >
    ZAP is partcipating in Hacktoberfest 2022.
images:
- https://www.zaproxy.org/site/blog/2022-10-01-zap-and-hacktoberfest/images/Hfest_Badge.png
type: post
tags:
- blog
- hacktoberfest
date: "2022-10-01"
addSocialPreview: true
authors:
- arkaprabha
---
![Hacktoberfest X ZAP](images/zap_Hacktoberfest.png)

We are happy to announce that ZAP is participating in [Hacktoberfest](https://hacktoberfest.com/) 2022.

## Contributors
This is what contributors need to know to participate in and complete Hacktoberfest 2022:

> - Go through the [Hacktoberfest 2022 contributor](https://hacktoberfest.com/participation/#contributors) section to know more about what you should do and the rewards you will recieve. We would be genuinely happy if you choose the "plant a tree" option as your reward.

Reviewer 2: recieve -> receive

- Contributions/pull requests can be made to the following repositories under the [OWASP ZAP](https://github.com/zaproxy) organization:
    - ZAP Core: [link](https://github.com/zaproxy/zaproxy)
    - ZAP Extensions: [link](https://github.com/zaproxy/zap-extensions)
    - ZAP HUD: [link](https://github.com/zaproxy/zap-hud)
    - ZAP Core Help: [link](https://github.com/zaproxy/zap-core-help)
    - ZAP Community Scripts: [link](https://github.com/zaproxy/community-scripts)

> - Read the [ZAP Contributing Guide](https://www.zaproxy.org/docs/contribute/).

Reviewer 2: ```suggestion
- Read the [ZAP Contributing Guide](/docs/contribute/).
```

- We recommend you to go through the [ZAP Developer Guide](/docs/developer/). It has all the answers to your questions about the development environment setup and the checks that your PR should pass before it is merged. 
- If you are a beginner, please check out the curated [Hacktoberfest issues](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3AHacktoberFest) on the main ZAP repo.
- If you would like to challenge yourself, you can check out the [good second issues](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+second+issue%22) and [enhancement](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) issues.

Some good resources to get started for beginners:
 - Open Source Guides: [How to contribute to Open Source](https://opensource.guide/how-to-contribute/)

>  - GitHub: [GitHub training kit](https://github.github.com/training-kit/)

Reviewer 2: ```suggestion
 - GitHub: [GitHub training kit](https://training.github.com/)
```

---

> title: "ZAP and Hacktoberfest"

Reviewer 2: Could include the year.

summary: >
    ZAP is partcipating in Hacktoberfest 2022.
images:

> - https://www.zaproxy.org/site/blog/2022-10-01-zap-and-hacktoberfest/images/Hfest-Badge-2-Color-Void@2x.png

Reviewer 3: I dont know if the '@' will cause problems with the social image?
It may well be fine but might be worth changing just in case as its really hard to test this before publishing 😉 
And we have had similar problems before..

Author: Yep sure... I just simoly need to change the filename :)

type: post
tags:
- blog
- hacktoberfest
date: "2022-10-01"
addSocialPreview: true
authors:
- arkaprabha
---
![Hacktoberfest X ZAP](images/zap_Hacktoberfest.png)

We are happy to announce that ZAP is participating in [Hacktoberfest](https://hacktoberfest.com/) 2022.

## Contributors

> This is what contributors need to know to participate and complete Hacktoberfest 2022:

Reviewer 2: ```suggestion
This is what contributors need to know to participate in and complete Hacktoberfest 2022:
```

- Register anytime between September 26 and October 31.
- Contributions/pull requests can be made to the following repositories under the [OWASP ZAP](https://github.com/zaproxy) organization:
    - ZAP Core: [link](https://github.com/zaproxy/zaproxy)
    - ZAP Extensions: [link](https://github.com/zaproxy/zap-extensions)

>     - ZAP Hud: [link](https://github.com/zaproxy/zap-hud)

Reviewer 4: HUD (all caps)

If we're really listing all of these.

    - ZAP Core Help: [link](https://github.com/zaproxy/zap-core-help)
    - ZAP Community Scripts: [link](https://github.com/zaproxy/community-scripts)
    - ZAP Website: [link](https://github.com/zaproxy/zaproxy-website)

>     - ZAP API Docs: [link](https://github.com/zaproxy/zap-api-docs)

Reviewer: What contributions are expected for these?

Author: Uhh, maybe grammatical corrections and detailed descriptions for certain areas?

Author: Any documentation issues or advanced examples :). Let's keep it open for contributors and see what they can add.. spam PRs would definitely be marked with a "spam" label.

Reviewer: Most of the content is generated for other repos/sources.

Author: agreed

Author: I saw two enhancement issues. So thought of including the repo link.

Reviewer: Which issues?

Author: [this](https://github.com/zaproxy/zap-api-docs/issues/88) and [this](https://github.com/zaproxy/zap-api-docs/issues/42).

Author: :)

Reviewer: Neither of those is a good first issue, both issues need to be updated to be more clear what's expected to be done.

Author: Yes ... Hacktoberfest doesn't necessarily revolve around good first.. literally, any issues can be solved and any PRs which aren't spammy and accepted with the label "hacktoberfest-accepted" counts as a submission.

Reviewer: It's about the cost benefit, leading contributors to issues that need more to time/effort from everyone is less useful than lead them to issues that are faster to understand and address.

Author: true that! I did mention that in the blog... "enhancement" issues will be challenging

Reviewer: "yourself" is not "everyone" though.

Author: Now I don't how to process this :)

Author: What should I do then? drop the link? :)

Reviewer: The link**s**, it's about both highlighted repos. I'd prefer that we focused the contributions to main repos.

- Read the [ZAP Contributing Guide](https://www.zaproxy.org/docs/contribute/).

> - We recommend you to go through [ZAP Development guide](https://www.zaproxy.org/docs/developer/), it has all the answers to your questions about the development environment setup and the verifications you must perform before your PR is merged. 

Reviewer 2: ```suggestion
- We recommend you to go through the [ZAP Developer Guide](/docs/developer/). It has all the answers to your questions about the development environment setup and the checks that your PR should pass before it is merged. 
```

> - If you are beginner, we recommend you to try out the curated [hacktoberfest issues](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3AHacktoberFest) we have for you. 

Reviewer 2: ```suggestion
- If you are a beginner, please check out the curated [Hacktoberfest issues](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3AHacktoberFest) on the main ZAP repo. 
```

> - If you would like to tackle a bit more challenging ones, you can check out the [good second issues](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+second+issue%22) and [enhancement](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) issues.

Reviewer 2: ```suggestion
- If you would like to challenge yourself, you can check out the [good second issues](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+second+issue%22) and [enhancement](https://github.com/zaproxy/zaproxy/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) issues.
```

- Have 4 pull requests accepted between October 1 and October 31.

> - The first 40,000 participants (maintainers and contributors) who successfully complete Hacktoberfest can elect to receive one out of two prizes: a tree planted in their name, or the Hacktoberfest 2022 t-shirt. We would greatly appreciate if you would pick the "plant a tree in your name" option.

Reviewer 2: We could add a single point which links to https://hacktoberfest.com/participation/#contributors and avoid repeating information here. This blog post could contain ZAP specific suggestions.

Author: Agreed! this would make the blog crisp and short.
